### ISSUE TITLE
Add new category controls improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#2572]
The mini-form to add a new category can be improved a bit for accessibility and better UI. In the screenshot below, the Gutenberg one (left) compared with the classic editor one (right):

![screen shot 2017-08-28 at 16 44 29](https://user-images.githubusercontent.com/1682452/29779078-c02b0542-8c11-11e7-9914-f4669ac6b973.png)

a11y/interaction regressions in the Gutenberg one:
- both the input field and the select element miss a properly associated label
- placeholder attributes should not be used as replacement for labels, see also https://core.trac.wordpress.org/ticket/40331
- the same "Add New Category" text (or "Add New Term") used for both the input field and the submit button can be potentially confusing
- the form shouldn't close when pressing the submit button
- the input field and the submit button miss a focus style (see screenshot below)

Further improvements worth considering:
- hide the "+" from assistive technologies, to avoid screen readers announce it as "plus"
- the toggle button needs an `aria-expanded` attribute to communicate the form expanded/collapsed state
- consider to use visible label elements for both the input and select
- consider to use "None" instead of "— Parent Category —" for the select first option
- consider to avoid using title casing for everything: "Add New Category" for the placeholder seems a bit excessive to me
- the submit button is unstyled (needs design feedback)

To clarify the missing focus style, in the screenshot below all the 3 elements (input, select, button) are focused using the Chrome inspector: please consider webkit on macOS adds its own native fous style (on the input) which is not the one we want:

<img width="281" alt="screen shot 2017-08-28 at 17 13 51" src="https://user-images.githubusercontent.com/1682452/29780042-afa54e78-8c14-11e7-9f27-605b16ef1e4d.png">

on Firefox, only the select element has a focus style:

<img width="279" alt="screen shot 2017-08-28 at 17 18 05" src="https://user-images.githubusercontent.com/1682452/29780145-f609d168-8c14-11e7-9317-1b8b7934d168.png">

**Comment by afercia** [comment]
Also (just noticed): elements that have focus should never gets disabled. See the submit button: when using a keyboard and activating the button, it gets disabled `disabled={ adding }` thus causing a focus loss (notice the form should stay open).

**Comment by afercia** [comment]
Further improvement:
- add `speak` message when a term is successfully added (maybe also on failure?)

**Comment by afercia** [comment]
The input field misses a "type" attribute...
